adapters/appnexus/appnexus.go

```diff
@@ -303,63 +304,61 @@ func makeKeywordStr(keywords []*openrtb_ext.ExtImpAppnexusKeyVal) string {
 }
 
 func (a *adapter) MakeBids(internalRequest *openrtb2.BidRequest, externalRequest *adapters.RequestData, response *adapters.ResponseData) (*adapters.BidderResponse, []error) {
-	if response.StatusCode == http.StatusNoContent {
+	if bidderutil.IsResponseCodeStatusNoContent(response) {
 		return nil, nil
 	}
 
-	if response.StatusCode == http.StatusBadRequest {
-		return nil, []error{&errortypes.BadInput{
-			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode),
-		}}
-	}
-
-	if response.StatusCode != http.StatusOK {
-		return nil, []error{fmt.Errorf("Unexpected status code: %d. Run with request.debug = 1 for more info", response.StatusCode)}
+	if err := bidderutil.CheckResponseStatusCodeForErrors(response); err != nil {
+		return nil, []error{err}
 	}
 
-	var bidResp openrtb2.BidResponse
-	if err := json.Unmarshal(response.Body, &bidResp); err != nil {
+	var appnexusResponse openrtb2.BidResponse
+	if err := json.Unmarshal(response.Body, &appnexusResponse); err != nil {
 		return nil, []error{err}
 	}
 
-	bidResponse := adapters.NewBidderResponseWithBidsCapacity(5)
-
 	var errs []error
-	for _, sb := range bidResp.SeatBid {
-		for i := 0; i < len(sb.Bid); i++ {
+	bidderResponse := adapters.NewBidderResponseWithBidsCapacity(5)
+	for _, sb := range appnexusResponse.SeatBid {
+		for i := range sb.Bid {
 			bid := sb.Bid[i]
+
 			var bidExt appnexusBidExt
 			if err := json.Unmarshal(bid.Ext, &bidExt); err != nil {
 				errs = append(errs, err)
-			} else {
-				if bidType, err := getMediaTypeForBid(&bidExt); err == nil {
-					if iabCategory, err := a.getIabCategoryForBid(&bidExt); err == nil {
-						bid.Cat = []string{iabCategory}
-					} else if len(bid.Cat) > 1 {
-						//create empty categories array to force bid to be rejected
-						bid.Cat = make([]string, 0)
-					}
-
-					impVideo := &openrtb_ext.ExtBidPrebidVideo{
-						Duration: bidExt.Appnexus.CreativeInfo.Video.Duration,
-					}
-
-					bidResponse.Bids = append(bidResponse.Bids, &adapters.TypedBid{
-						Bid:          &bid,
-						BidType:      bidType,
-						BidVideo:     impVideo,
-						DealPriority: bidExt.Appnexus.DealPriority,
-					})
-				} else {
-					errs = append(errs, err)
-				}
+				continue
+			}
+
+			bidType, err := getMediaTypeForBid(&bidExt)
+			if err != nil {
+				errs = append(errs, err)
+				continue
 			}
+
+			iabCategory, found := a.findIabCategoryForBid(&bidExt)
+			if found {
+				bid.Cat = []string{iabCategory}
+			}
+
+			if len(bid.Cat) > 1 {
```

Comment:
can we keep this if statement as the else part of `if found` statement as was done in the previous code

Reply:
https://github.com/prebid/prebid-server/pull/2638/commits/9f17bab35ba64c9c810c4040794e7c6c6c6a67d5